### ISSUE TITLE
Improvement: Make store hydrate to localstorage if needed

### ISSUE THREAD

**Issue by eznix86** [flawiddsouza/Restfox#108]
- Currently if we need to store elements into store and persist them, we have to do it in 2 steps.
- Which is storing the data to local storage then retrieve the data.

Instead we can do,
- On mutation of the store we save to local storage (can be replaced with dixiejs because local storage is synchronous).
- On App boot we hydrate the data in to the state.

Example here:
https://www.mikestreety.co.uk/blog/vue-js-using-localstorage-with-the-vuex-store/

```
app <-> state <-> update storage on mutation / fetch state on boot.
```

Benefits:
- Reduce checks,
- storage is coupled/bounded to the local storage,
- Less code to maintain.

**Comment by flawiddsouza** [comment]
Sounds like a good improvement. But we don't need to persist everything in the store to the localStorage. If we can limit it to the items that are used for configuring the application, that would be better. I've been planning to move away from localStorage but I never got the time to do it. This would mean we'd have to migrate the existing user configuration as well, as we can't have users suddenly losing their existing settings because there's a new update to the settings system.

**Comment by eznix86** [comment]
I agree, but I think we can prevent a breaking change if we do it meticulously. 

Just proposing to make it simpler because when coding on the previous PR, It seems weird to always update the storage then change the store.
